sky/clouds/service_catalog/gcp_catalog.py

```diff
@@ -332,6 +332,14 @@ def list_accelerators(
                                             region_filter, quantity_filter,
                                             case_sensitive)
 
+    # Remove GPUs that are unsupported by SkyPilot.
+    results = {
+        k: v
+        for k, v in results.items()
+        if (k in list(_NUM_ACC_TO_MAX_CPU_AND_MEMORY.keys()) +
+            list(_A100_INSTANCE_TYPE_DICTS.keys()) or 'tpu' in k.lower())
+    }
```

Comment:
style nit: I think the code is not easy to read. Why don't we use a plain for loop instead of the dictionary comprehension?

Reply:
Good point! Fixed. Thanks!